From bc8af42f4f04a9853269aa446616f43f9a944882 Mon Sep 17 00:00:00 2001
From: Vladimir Oltean <olteanv@gmail.com>
Date: Thu, 27 Sep 2018 01:33:04 +0300
Subject: [PATCH 11/18] common: create proc_kernel_sched domain to restrict
 perf hal access

* Also used to limit vendor_init (init.qcom.power.rc) access to /proc

Change-Id: I3027ce621c62d3fc9ad03b6d79554d4ac2e0ec89
Signed-off-by: Vladimir Oltean <olteanv@gmail.com>
---
 common/file.te             |  3 +++
 common/genfs_contexts      | 14 ++++++++++++++
 common/hal_perf_default.te |  2 +-
 common/vendor_init.te      |  1 +
 4 files changed, 19 insertions(+), 1 deletion(-)
 create mode 100644 common/vendor_init.te

diff --git a/common/file.te b/common/file.te
index 32d6053..0522f12 100644
--- a/common/file.te
+++ b/common/file.te
@@ -46,6 +46,9 @@ type diag_data_file, file_type, data_file_type;
 #file type for restricting proc read by audiod
 type proc_audiod, fs_type;
 
+#file type for restricting init.qcom.power.rc
+type proc_kernel_sched, fs_type;
+
 #file type for irqbalance socket
 type msm_irqbalance_socket, file_type;
 
diff --git a/common/genfs_contexts b/common/genfs_contexts
index e57b6fe..f96b3b4 100644
--- a/common/genfs_contexts
+++ b/common/genfs_contexts
@@ -20,3 +20,17 @@ genfscon sysfs /devices/virtual/graphics u:object_r:sysfs_graphics:s0
 genfscon sysfs /devices/virtual/thermal u:object_r:sysfs_thermal:s0
 genfscon sysfs /devices/virtual/kgsl/kgsl/proc u:object_r:sysfs_kgsl_proc:s0
 genfscon debugfs /kgsl/proc     u:object_r:kgsl_debugfs:s0
+genfscon proc  /sys/kernel/sched_boost                  u:object_r:proc_kernel_sched:s0
+genfscon proc  /sys/kernel/sched_upmigrate              u:object_r:proc_kernel_sched:s0
+genfscon proc  /sys/kernel/sched_downmigrate            u:object_r:proc_kernel_sched:s0
+genfscon proc  /sys/kernel/sched_grp_upmigrate          u:object_r:proc_kernel_sched:s0
+genfscon proc  /sys/kernel/sched_grp_downmigrate        u:object_r:proc_kernel_sched:s0
+genfscon proc  /sys/kernel/sched_window_stats_policy    u:object_r:proc_kernel_sched:s0
+genfscon proc  /sys/kernel/sched_enable_thread_grouping u:object_r:proc_kernel_sched:s0
+genfscon proc  /sys/kernel/sched_freq_inc_notify        u:object_r:proc_kernel_sched:s0
+genfscon proc  /sys/kernel/sched_freq_dec_notify        u:object_r:proc_kernel_sched:s0
+genfscon proc  /sys/kernel/sched_ravg_hist_size         u:object_r:proc_kernel_sched:s0
+genfscon proc  /sys/kernel/sched_init_task_load         u:object_r:proc_kernel_sched:s0
+genfscon proc  /sys/kernel/sched_spill_nr_run           u:object_r:proc_kernel_sched:s0
+genfscon proc  /sys/kernel/sched_small_task             u:object_r:proc_kernel_sched:s0
+
diff --git a/common/hal_perf_default.te b/common/hal_perf_default.te
index 1a2234c..d721a1a 100644
--- a/common/hal_perf_default.te
+++ b/common/hal_perf_default.te
@@ -40,7 +40,7 @@ add_hwservice(hal_perf_server, hal_perf_hwservice)
 allow hal_perf_client hal_perf_hwservice:hwservice_manager find;
 
 allow hal_perf cgroup:file r_file_perms;
-allow hal_perf_default proc:file rw_file_perms;
+allow hal_perf_default proc_kernel_sched:file rw_file_perms;
 allow hal_perf device_latency:chr_file rw_file_perms;
 allow hal_perf_default mpctl_data_file:dir rw_dir_perms;
 allow hal_perf_default mpctl_data_file:file create_file_perms;
diff --git a/common/vendor_init.te b/common/vendor_init.te
new file mode 100644
index 0000000..e6ea53c
--- /dev/null
+++ b/common/vendor_init.te
@@ -0,0 +1 @@
+allow vendor_init proc_kernel_sched:file w_file_perms;
-- 
2.17.1

